packages/web/server/queries/complex/assets/index.ts

```diff
@@ -0,0 +1,118 @@
+import { Asset } from "@osmosis-labs/types";
+import { getDisplayDecimalsFromAsset } from "@osmosis-labs/utils";
+import cachified, { CacheEntry } from "cachified";
+import Fuse from "fuse.js";
+import { LRUCache } from "lru-cache";
+
+import { DEFAULT_LRU_OPTIONS } from "~/config/cache";
+import { AssetLists } from "~/config/generated/asset-lists";
+
+import { Search, Sort } from "../parameter-types";
+
+export type GetAssetsParams = {
+  sort?: Partial<Sort>;
+  search?: Search;
+  /** Explicitly match the base or symbol denom. */
+  matchDenom?: string;
+};
+const searchableKeys = ["symbol", "base", "name", "display"];
+
+const cache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
+
+/** Get an individual asset explicitly by it's denom (any type) */
+export async function getAsset(
+  anyDenom: string
+): Promise<ReturnType<typeof makeMinimalAsset> | undefined> {
+  const assets = await getAssets({ matchDenom: anyDenom });
+  return assets[0];
+}
+
+/** Cached function that returns minimal asset information. Return values can double as the `Currency` type.
+ *  Please avoid adding to this function unless absolutely necessary.
+ *  Instead, compose this function with other functions to get the data you need.
+ *  The goal is to keep this function simple and lightweight. */
+export async function getAssets(
+  params: GetAssetsParams,
+  assetList = AssetLists
```

Comment:
What are your thoughts on placing assetlist within the params? 

```suggestion
  { ...params, assetList }: GetAssetsParams & { assetList: AssetList[] },
```